### ISSUE TITLE
Internal error in syntax error in example with floating point value

### ISSUE THREAD

**Issue by matszpk** [golang/go#36670]
Error encountered while trial of building some trivial example:
```
package main

import "fmt"

type Ab struct {
    v float64
    i int
}

func main() {
    t := Ab{ 144&.0, 445 }
    fmt.Println(t)
}
```

Compiler (go clang 1.12.12) just prints following error:
```
# command-line-arguments
./internal-error.go:11:17: internal compiler error: binaryOp: bad operation: 144 & 0

Please file a bug report including a short program that triggers the error.
https://golang.org/issue/new
```

I reproduced this bug on a official version go 1.12.12 (clang). gccgo print correct error that describes syntax error:

```
./internal-error.go:11:17: error: expected integer type
   11 |     t := Ab{ 144&.0, 445 }
      |
```

Go version: go version go1.12.12 linux/amd64

GO env:
```
GOARCH="amd64"
GOBIN=""
GOCACHE="/mnt/oldsys/home/mat/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/mat/docs/dev/go/c00lpool-ws"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="/usr/bin/gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build945722657=/tmp/go-build -gno-record-gcc-switches"
```

**Comment by ALTree** [comment]
Hi,

thanks for reporting this.

This is a dup of #31060, already fixed in 1.13 and on tip. 

Closing here, since no further action is required.

**Comment by ALTree** [comment]
> Compiler (go clang 1.12.12)

(and btw the official compiler is not based on clang). I assume you mean gc, the compiler distributed on golang.org.
